src/graph.rs

```diff
@@ -559,12 +613,8 @@ pub struct TypesDependency {
 #[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq)]
 #[serde(rename_all = "camelCase")]
 pub enum ModuleKind {
-  /// An asserted module. The import location is required to determine what the
-  /// asserted type is as well as a loader/runtime would want to ensure the
-  /// `MediaType` matches the assertion. Dependency analysis does not occur on
-  /// asserted modules.
-  Asserted,
```

Comment:
I've merged `ModuleKind::Asserted` into `ModuleKind::Esm`. That info is given by `Module::media_type` and assert-worthy module types should fall in naturally for other purposes. The name seems more related to module system not language so it should be fine.

Reply:
I'm doing a large refactor for npm specifiers atm that's going to remove `ModuleKind` and replace it with an `enum Module`. We're not going to land this PR until deno 2.0 so you'll get some conflicts on this on this one, but I don't think it will be too much trouble.

Reply:
I think we could represent it as `Module::Json` instead of `Module::Asserted`. I'll update my PR to that.